cmd/agent/daemon/state/events_pipeline.go

```diff
@@ -4,71 +4,46 @@ import (
 	"bytes"
 	"context"
 	"encoding/json"
-	"fmt"
+	"net/netip"
 
 	castpb "github.com/castai/kvisor/api/v1/runtime"
 	"github.com/castai/kvisor/cmd/agent/daemon/enrichment"
-	"github.com/castai/kvisor/pkg/castai"
 	ebpftypes "github.com/castai/kvisor/pkg/ebpftracer/types"
 	"github.com/castai/kvisor/pkg/metrics"
-	"github.com/prometheus/client_golang/prometheus"
-	"google.golang.org/grpc"
-	"k8s.io/apimachinery/pkg/types"
 )
 
-// runEventsExportLoop sends events to server.
-func (c *Controller) runEventsExportLoop(ctx context.Context) error {
-	c.log.Info("running events sink loop")
-	defer c.log.Info("events sink loop done")
-
-	ws := castai.NewWriteStream[*castpb.Event, *castpb.WriteStreamResponse](ctx, func(ctx context.Context) (grpc.ClientStream, error) {
-		return c.castClient.GRPC.EventsWriteStream(ctx)
-	})
-	defer ws.Close()
-	ws.ReopenDelay = c.writeStreamCreateRetryDelay
+func (c *Controller) runEventsPipeline(ctx context.Context) error {
+	c.log.Info("running events pipeline")
+	defer c.log.Info("events pipeline done")
 
 	for {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
-		case e := <-c.eventsExportQueue:
-			c.enrichEvent(e)
-			if err := ws.Send(e); err != nil {
+		case e := <-c.tracer.Events():
+			pbEvent := c.toProtoEvent(e)
```

Comment:
Note that I removed event enrich from here. Since we need more data on backend anyway we can safe some bytes and skip sending workload info here.